main.c

```diff
@@ -0,0 +1,200 @@
+            }
+            else{
+                WHITE_LED_ON();
+                on = 1;
+            }
+            
+            can_msg_t alt_1_arm_stat_msg;
+            build_arm_stat_msg(millis(), 
+                               1, 
+                               alt_1_arm_state,
+                               (uint16_t)ADCC_GetSingleConversion(A1_DROGUE_PIN)*3.72,
+                               (uint16_t)ADCC_GetSingleConversion(A1_MAIN_PIN)*3.72,
+                               &alt_1_arm_stat_msg);
+            txb_enqueue(&alt_1_arm_stat_msg);
+            
+            can_msg_t alt_2_arm_stat_msg;
+            build_arm_stat_msg(millis(), 
+                               2, 
+                               alt_2_arm_state,
+                               (uint16_t)(ADCC_GetSingleConversion(A2_DROGUE_PIN))*3.72,
+                               (uint16_t)(ADCC_GetSingleConversion(A2_MAIN_PIN))*3.72,
+                               &alt_2_arm_stat_msg);
+            txb_enqueue(&alt_2_arm_stat_msg);
+            
+            can_msg_t bat_1_v_msg;
+            build_analog_data_msg(millis(), 
+                                  SENSOR_ARM_BATT_1, 
+                                  (uint16_t)(ADCC_GetSingleConversion(BATTERY_1_PIN))*3.72,
+                                  &bat_1_v_msg);
+            txb_enqueue(&bat_1_v_msg);
+            
+            can_msg_t bat_2_v_msg;
+            build_analog_data_msg(millis(), 
+                                  SENSOR_ARM_BATT_2, 
+                                  (uint16_t)(ADCC_GetSingleConversion(BATTERY_2_PIN))*3.72,
+                                  &bat_2_v_msg);
+            txb_enqueue(&bat_2_v_msg);
+            
+        }
+        
+        parse_altitude();
+        if (new_altitude_available()){
+            can_msg_t altitude_msg;
+            build_altitude_data_msg(millis(), get_altitude(), &altitude_msg);
+            txb_enqueue(&altitude_msg);
+        }
+        // set io to arm state of altimeter 1
+        if(alt_1_arm_state == DISARMED) {
+            DISARM_A1();
+            RED_LED_OFF();
+        }
+        else {
+            ARM_A1();
+            RED_LED_ON();
+        }
+        
+        // set io to arm state of altimeter 2
+        if(alt_2_arm_state == DISARMED) {
+            DISARM_A2();
+            BLUE_LED_OFF();
+        }
+        else {
+            ARM_A2();
+            BLUE_LED_ON();
+        }
+        
+        // send queued messages
+        txb_heartbeat();
+    }
+    return (EXIT_SUCCESS);
+}
+
+static void __interrupt() interrupt_handler(){
+    if (PIE3bits.TMR0IE == 1 && PIR3bits.TMR0IF == 1) {
+        timer0_handle_interrupt();
+        PIR3bits.TMR0IF = 0;
+    }
+    if (PIR5) {
+        can_handle_interrupt();
+    }
+    if(U1ERRIRbits.FERIF == 1 || U1ERRIRbits.RXFOIF){   //should probably do something if there is an error
+        RED_LED_ON();
+    }
+    else if (PIE3bits.U1RXIE == 1 && PIR3bits.U1RXIF == 1){
```

Comment:
You don't need to check if `PIE3bits.U1RXIE` is set. That's just checking if you've enabled UART receive interrupts. If you haven't enabled them, the interrupt flag wouldn't be set, so checking the enable bit doesn't add anything.

Checking the RX flag only is sufficient.